pkg/strategy/controller.go

```diff
@@ -79,14 +79,20 @@ func normalize(ctx context.Context, jaeger *v1.Jaeger) {
 	}
 
 	// remove reserved labels
-	delete(jaeger.Spec.JaegerCommonSpec.Labels, "app.kubernetes.io/instance")
-	delete(jaeger.Spec.JaegerCommonSpec.Labels, "app.kubernetes.io/managed-by")
-
-	delete(jaeger.Spec.AllInOne.JaegerCommonSpec.Labels, "app.kubernetes.io/instance")
-	delete(jaeger.Spec.AllInOne.JaegerCommonSpec.Labels, "app.kubernetes.io/managed-by")
-
-	delete(jaeger.Spec.Query.JaegerCommonSpec.Labels, "app.kubernetes.io/instance")
-	delete(jaeger.Spec.Query.JaegerCommonSpec.Labels, "app.kubernetes.io/managed-by")
+	for _, labels := range []map[string]string{
+		jaeger.Spec.JaegerCommonSpec.Labels,
+		jaeger.Spec.AllInOne.JaegerCommonSpec.Labels,
+		jaeger.Spec.Query.JaegerCommonSpec.Labels,
+	} {
+		if _, ok := labels["app.kubernetes.io/instance"]; ok {
+			jaeger.Logger().Warnf("The reserved label 'app.kubernetes.io/instance' is overwritten. Falling back to %s", jaeger.Name)
```

Comment:
Sorry for being picky, but it really belongs as a span event. A span is created on line 58, you can use it here. Unfortunately, we do not have a good way to send back this kind of information to the user who created the CR (event, status message, ...) in this code, as we have in the upgrade package, and leaving log messages at the operator stdout would likely only pollute the logs. Logging to the span would allow people looking at the trace for a reconciliation to see this in connection to the bigger context.